File: TextIndexApi/src/main/java/com/hideakin/textsearch/index/model/TextToken.java
package com.hideakin.textsearch.index.model;

public class TextToken {

	private String text;
	
	private int row;
	
	private int col;
	
	public TextToken(String text, int row, int col) {
		this.text = text;
		this.row = row;
		this.col = col;
	}

	public String getText() {
		return text;
	}

	public int getRow() {
		return row;
	}

	public int getCol() {
		return col;
	}

}

File: TextSearchCLI/model/TextToken.cs
namespace com.hideakin.textsearch.model
{
    /// <summary>
    /// A token is defined as the longest possible character sequence that do not contain any embedded delimiters.
    /// </summary>
    internal class TextToken
    {
        public string Text { get; }

        public int Row { get; }

        public int Column { get; }

        public TextToken(string text, int row, int column)
        {
            Text = text;
            Row = row;
            Column = column;
        }
    }
}
